### ISSUE TITLE
Images Not Found

### ISSUE THREAD

**Issue by programetica** [linnovate/mean-upload#6]
Hi Lior, thanks for the great package!  I've been trying to figure out how to implement something similar for two months (I'm new to node).  Everything installed and ran perfectly, except one minor detail I'm trying to figure out.  My images don't show the way they do in your video.  If I look in the console, I see "Failed to load resource:  the server responded wiht a status of 404 (Not Found)".  The code also doesn't generate anything on the example page past "The example above generates the following json which can be used by your own package:"  It shows the right URL for where the image was uploaded though.  Thanks for any input!

**Comment by adle29** [comment]
Yes, that happened to me too! What I did, was to save the images inside "/articles/assets/img/" and there it was the only place I was able to access the images.

**Comment by programetica** [comment]
Thanks so much for replying to this with your solution, I can't wait to try that out!

**Comment by liorkesos** [comment]
I'll try to o fix the defaults to work as well.

**Comment by programetica** [comment]
Thanks, Lior!

**Comment by OscarGold** [comment]
I have the same issue. I have noticed that the client does successfully get the file (from its local disk or whatever) and successfully sends it to the server. The 404 error occurs when it (the server) tries to send the file back (to the client) for the carosel. Its seems its actually the server that cannot see the file in its own directories. I even tried the dir mentioned above, and several other places that had similar files.
I am not sure, but it would appear that maybe the file needs to be validated somehow (by what, I dont know). I cant workout where the express static dir is either (if there is one), that would seem an ideal use...
BTW  LOVE MEAN.IO. Well built, extremely flexible and extensive. Best Node boilerplate/Template by a country mile.. IMHO. Thanks..... 
also... what is PRS?

**Comment by programetica** [comment]
Thanks for the detailed account, Oscar.  I'm wondering in my case if it had anything to do with my using the code on one computer only rather than actually setting it up on a server and accessing it from a client machine?  Am I trying to violate cross-domain security (like by accessing a folder on the "client machine" from the "server" -- which in my case are the same machine) or something?

Lior, can you mention how you have it set up so that others can try to replicate it?

Thanks!

**Comment by jmckenney** [comment]
You likely need to tell express/node how to deliver it. I had the same issue where it was working locally but not on aws. Until I set this in app.js in the package:

```
app.use('/uploads/locations', express.static(__dirname + '/public/assets/static'));
```

That sets /uploads/locations/[file_name.xxx] to be available even though it was saved in [name_of_package]/public/assets/static/[file_name.xxx].

My hunch is that it was working locally because my node or express was set to deliver all static files. The truth is we shouldn't be trying to serve static files via node but should rather build in a cdn or serve via nginx or apache. But for now, this express setting will work for us:)

**Comment by liorkesos** [comment]
I'll try to set up a reference implementation...
Lior

**Comment by OscarGold** [comment]
Thanks jm and thanks lior. I have been going through the code.. I noticed line 65 of express.js was..
"    webroot: /public\/|packages\//g" which is an optional parameter to truncate the directory name reported at the console (dev mode obviously). After I got rid of that line, the console started making alot more sense.
@holodny I have not encountered any cross domain issues... and @jmckenney is on the right track, and agree re avoiding static files if possible. Apart from everything else, they are never cached by Node,(so I heard)  which would cause unnecessary load on a busy server.

Although I am still puzzled that the server seems not to be able to find it in its own local dir, and Im not sure that serving it to itself will help.. my logic being :if it can see it to serve it, then it doesn't need to serve it(to itself).. Comms from the client (which is the hard part) seem OK. 

I have a hunch it may have something to with compression (or whatever catches it and processes it when it pops out the angular.js pipe that the client sent it through.)
One interesting point is that its seems to be sharing (something... code... dir space) with the Articles plugin (if there is one).. could be a red herring... dunno.. just a hunch.

I must say again, this is a relatively trivial problem (as evidenced by the fact Im getting close to understanding it), most other boiler plate/templates I have seen have many more profoundly more complex issues, BEFORE randomly installing several random plugins. I am enjoying going through the code and working it out. It is beautifully laid out, and very well commented. Kudos to you and the team @liorkesos :)

**Comment by OscarGold** [comment]
What I have found:
The file uploaded by the client is successfully passed to and written  (on the server) by the 
node_modules\mean-upload\server\controllers\meanUpload.js file.
The file is created by the fs.rename command (line 10), and its rights are granted.
The first thing you notice is that the word "Creator" is spelt wrong "createor", but changing that changed nothing.
The file appears in the windows directory  with a padlock icon on it. 
This means it does not have full NTFS rights. These can be restored by giving access to"Authenticated Users" or "Users" Groups manually, but by then its too late for node.

This link sort of describes it..
http://www.howtogeek.com/howto/17117/remove-the-lock-icon-from-a-folder-in-windows-7/

I could probably change the fs.rename command or add a command to assign the correct rights or change the rights after the fact, but I'm too much of a noob at Node to do that just yet....
I did get the error to disappear by removing lines 18 through 24. This stopped the 404 error, but then the Carosel function does seem to be running at all (it reports 0 files uploaded)
any ideas?

**Comment by OscarGold** [comment]
Just an update to that last bit...
 Found that the file was first being written to the OS\TEMP directory which had the limited permissions before being moved to the "files-dest" folder. 
Changed the rights on the temp folder to have proper rights, and now the file shows up without the padlock icon on it. (yay!)
But alas the 404 continues :) darn. wild goose chase. 
Back to the drawing board.
P.S. sorry about the wordy posts... tell me to shut up if Im writing too much :)

**Comment by liorkesos** [comment]
Well ha some funky git issues so could not fix it directly in the package.
It worked for me after changing the dst when invoking the upload directive
in the view to - <mean-upload file-dest="'/packages/system/public/files/'"
and changing the meanUpload.js controller in around line 18 to -

 $scope.addSlide('/system/files/' + file.name); instead of
$scope.addSlide(file.src);

I'll try to get something out tommorow.

Lior

On Sun, Sep 7, 2014 at 10:36 PM, Oscar Goldman notifications@github.com
wrote:

> Just an update to that last bit...
> Found that the file was first being written to the OS\TEMP directory which
> had the limited permissions before being moved to the "files-dest" folder.
> Changed the rights on the temp folder to have proper rights, and now the
> file shows up without the padlock icon on it. (yay!)
> But alas the 404 continues :) darn. wild goose chase.
> Back to the drawing board.
> P.S. sorry about the wordy posts... tell me to shut up if Im writing too
> much :)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/linnovate/mean-upload/issues/6#issuecomment-54757731.

## 

Lior Kesos - http://www.linnovate.net
Linnovate - Community Infrastructure Care
mail: lior@linnovate.net
office: +972 722500881
cell: +972 524305252
skype: liorkesos

**Comment by OscarGold** [comment]
OMG THAT WORKED!!!
Thanks very Much!
Great job.

**Comment by liorkesos** [comment]
Now lets make it work by installing the package :)
Lior

On Mon, Sep 8, 2014 at 11:56 AM, Oscar Goldman notifications@github.com
wrote:

> OMG THAT WORKED!!!
> Thanks very Much!
> Great job.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/linnovate/mean-upload/issues/6#issuecomment-54791794.

## 

Lior Kesos - http://www.linnovate.net
Linnovate - Community Infrastructure Care
mail: lior@linnovate.net
office: +972 722500881
cell: +972 524305252
skype: liorkesos

**Comment by OscarGold** [comment]
Yes Cool.
I just noticed that Im getting the same issue when trying to load a file (as opposed to an image) from the link created.

Scratch that - I solved it by changing mean-upload/public/views/index.html line

"'  data-ng-href="{{file.src}}">{{file.name}} is {{file.size}} bytes     '" to 
"'     data-ng-href="{{'/system/files/' + file.name}}">{{file.name}} is {{file.size}} bytes  '"

And the LINK TO THE FILE WORKS ALSO !!!!

Whats a PRS? If I can return the favour by doing some stuff, I would be happy to.
